src/store/sagas/genres.js

```diff
@@ -0,0 +1,42 @@
+import {
+  all, call, put, takeLatest,
+} from 'redux-saga/effects';
+import genresApi from '../../api/genres';
+import {
+  GET_GENRES_PENDING,
+  GET_GENRES_BY_ID_PENDING,
+  GET_GENRES_SUCCESS,
+  GET_GENRES_ERROR,
+} from '../actionTypes';
+
+function* loadGenres() {
+  try {
+    const genres = yield call(genresApi.get);
+    yield put({
+      type: GET_GENRES_SUCCESS,
+      payload: { genres },
+    });
+  } catch (e) {
+    yield put({ type: GET_GENRES_ERROR, payload: e.message });
+  }
+}
+
+function* loadGenresById(action) {
+  const { ids } = action.payload;
+  try {
+    const genres = yield all(ids.map((id) => call(genresApi.getById, id)));
+    yield put({
+      type: GET_GENRES_SUCCESS,
```

Comment:
please, use `GET_GENRES_BY_IDS_SUCCESS` here and in reducer add genres to store (not completely override)